packages/typespec-client-generator-core/test/decorators.test.ts

```diff
@@ -2969,4 +2969,139 @@ describe("typespec-client-generator-core: decorators", () => {
       ok(testModel);
     });
   });
+
+  describe("@alternateType", () => {
+    describe.each([
+      ["utcDateTime", "string"],
+      ["utcDateTime", "int64"],
+      ["duration", "string"],
```

Comment:
there is a logic to add encoding for model property. see [here](https://github.com/Azure/typespec-azure/blob/111e0b9603c0b2b28415cbecd8d9669945f8b4c7/packages/typespec-client-generator-core/src/types.ts#L1127). could you add some tests to check if it will broke?

Reply:
added logic and unit tests for `@encode` support